programs/vote/src/vote_processor.rs

```diff
@@ -153,7 +154,23 @@ pub fn process_instruction(
             } else {
                 None
             };
-            vote_state::withdraw(me, lamports, to, &signers, rent_sysvar.as_deref())
+            let reject_vote_account_close_feature_active = invoke_context
+                .feature_set
+                .is_active(&feature_set::reject_vote_account_close_unless_zero_credit_epoch::id());
+
+            let clock = &from_keyed_account::<Clock>(keyed_account_at_index(
```

Comment:
I copied this logic from [the VoteInstruction::AuthorizeChecked just below it](https://github.com/solana-labs/solana/blob/master/programs/vote/src/vote_processor.rs#L172). It seems valid here but I'm not confident in it.

Reply:
This would require callers to have to include the Clock sysvar in the instruction accounts. It used to be that sysvars did need to be provided that way (and there's a lot of legacy code of that ilk in the Stake and Vote programs). But no longer!
Sysvars are now available as syscalls within InvokeContext; that's what the `reject_non_rent_exempt_vote_withdraws` feature is doing with the Rent sysvar just above. Here's an example of getting the Clock sysvar: https://github.com/solana-labs/solana/blob/c0638439be313efc6e0984aff6bdfd794dfa1713/programs/stake/src/stake_instruction.rs#L192
We can probably copy the `reject_non_rent_exempt_vote_withdraws` approach and just pass a single `Option<Clock>` into `VoteState::withdraw()` instead of adding 2 params.